### ISSUE TITLE
unable to restore backup

### ISSUE THREAD

**Issue by JamesBradyBear** [dtag-dev-sec/tpotce#298]
Hi,
I am trying to backup to backup data from cloud to local VM, the following information is regarding to Cloud VM, I also have same VM locally. I used the script `dump_es.sh` and it created the backup file which I downloaded it and I used it with restore_es.sh but it is failed to do so.

![image](https://user-images.githubusercontent.com/1931778/52886673-5811b180-3175-11e9-950f-6eeb0ab1f28c.png)

I have also tried backing up kibana objects and importing them but they also fail because they cannot find a folder also same case with backup created with `backup_es_folder.sh`

It has destroyed indexes if I'll go to es head, correct indexes are in following picture that how they are looking in the cloud 

![image](https://user-images.githubusercontent.com/1931778/52887871-b0e34900-3179-11e9-851a-5d3a6847e32d.png)

This is how they are looking locally 

![image](https://user-images.githubusercontent.com/1931778/52888945-1df8dd80-317e-11e9-945d-a00bf4fc35cf.png)

Running this command following is the output 
`docker logs logstash --follow`
![image](https://user-images.githubusercontent.com/1931778/52887959-16373a00-317a-11e9-86fb-9cdf597b9ed0.png)

![image](https://user-images.githubusercontent.com/1931778/52888763-4b915700-317d-11e9-96b6-473adcbfec93.png)

* What T-Pot version are you currently using? **18.11**
* Are you running on a Intel NUC or a VM? **VM**
* How long has your installation been running? **2 hours** on new VM
* Did you install any upgrades or packages? **YES** only linux updates before installing Tpot
* Did you modify any scripts? **Yes**
  `/opt/tpot/etc/logrotate`
  `/opt/tpot/etc/curator/actions.yml`
  Increased logging duration as explained here [issue #287](https://github.com/dtag-dev-sec/tpotce/issues/287)
* Have you turned persistence on/off? **I am unable to find persistence.on or off file so I followed this post** [Issue 287](https://github.com/dtag-dev-sec/tpotce/issues/287)
* How much RAM is available (login via ssh and run `htop`)? **6G / 7.79G**
* How much stress are the CPUs under (login via ssh and run `htop`)? **21%** Total 27% from hosting manager information
* How much swap space is being used (login via ssh and run `htop`)? **0**
* How much free disk space is available (login via ssh and run `sudo df -h`)? **35% used out of 160 GB**
* What is the current container status (login via ssh and run `sudo dps.sh`)?
```
NAME                  STATUS                       PORTS
adbhoney              Up 8 hours             0.0.0.0:5555->5555/tcp
ciscoasa              Up 8 hours
conpot_guardian_ast   Up 8 hours             0.0.0.0:10001->10001/tcp
conpot_iec104         Up 8 hours             0.0.0.0:161->161/tcp, 0.0.0.0:2404->2404/tcp
conpot_ipmi           Up 8 hours             0.0.0.0:623->623/tcp
conpot_kamstrup_382   Up 8 hours             0.0.0.0:1025->1025/tcp, 0.0.0.0:50100->50100/tcp
cowrie                Up 8 hours             0.0.0.0:22-23->22-23/tcp
cyberchef             Up 8 hours (healthy)   127.0.0.1:64299->8000/tcp
dionaea               Up 8 hours
elasticpot            Up 8 hours             0.0.0.0:9200->9200/tcp
elasticsearch         Up 8 hours (healthy)   127.0.0.1:64298->9200/tcp
ewsposter             Up 8 hours
glutton               Up 8 hours
head                  Up 8 hours (healthy)   127.0.0.1:64302->9100/tcp
heralding             Up 8 hours             0.0.0.0:110->110/tcp, 0.0.0.0:143->143/tcp, 0.0.0.0:993->993/tcp, 0.0.0.0:995->995/tcp, 0.0.0.0:5432->5432/tcp, 0.0.0.0:5900->5900/tcp
kibana                Up 8 hours (healthy)   127.0.0.1:64296->5601/tcp
logstash              Up 8 hours (healthy)
mailoney              Up 8 hours             0.0.0.0:25->25/tcp
medpot                Up 8 hours             0.0.0.0:2575->2575/tcp
nginx                 Up 8 hours
p0f                   Up 8 hours
rdpy                  Up 8 hours             0.0.0.0:3389->3389/tcp
snare                 Up 8 hours             0.0.0.0:80->80/tcp
spiderfoot            Up 8 hours (healthy)   127.0.0.1:64303->8080/tcp
suricata              Up 8 hours
tanner                Up 8 hours
tanner_api            Up 8 hours
tanner_phpox          Up 8 hours
tanner_redis          Up 8 hours             6379/tcp
tanner_web            Up 8 hours
```

Thanks.

**Comment by JamesBradyBear** [comment]
ok after some time (an hour maybe) they automatically started running but indexes are still corrupted.

**Comment by t3chn0m4g3** [comment]
Some things within Kibana and ES have changed. For now please use elastic dump directly. Follow the backup scripts as orientation on how to use it.

**Comment by JamesBradyBear** [comment]
Thank you very much for your reply. I'll use `dump_es.sh` and I'll try to restore it on another VM with same configurations. I tried it before but it failed and destroyed all the indexes, I'll do it again and document the output so that will be helpful as debug information for your as a help while fixing the issue. 

Thanks a lot for replying me on weekends :-)

**Comment by JamesBradyBear** [comment]
Hello,
I have created another VM with 16 GB or RAM, everything is superior than the previous one and tried to restore the backup again. 
Here is the result
![image](https://user-images.githubusercontent.com/1931778/52970511-4e29c180-33b4-11e9-8476-a7bed84ed426.png)
It threw an error but this time it didn't destroyed the indexes
![image](https://user-images.githubusercontent.com/1931778/52970532-66014580-33b4-11e9-9416-2dc9fbd4e84e.png)

After that I rebooted the system just to be sure, Tpot was not starting so I have to manually start the service and after that everything seems fine. Indexes are not destroyed and I can see data in Kibana.

Thanks, I hope that was helpful.

**Comment by t3chn0m4g3** [comment]
Thank you.

**Comment by t3chn0m4g3** [comment]
As far as I can test this is fixed for [19.03](https://github.com/dtag-dev-sec/tpotce/releases/tag/19.03.beta). Please have a test drive and let me know should you have still problems with dump / restore.

Thanks for bringing this to my attention :bowtie:
